Validation/RecoParticleFlow/plugins/PFClusterValidation.cc

```diff
@@ -1,271 +1,233 @@
+}
 
 void PFClusterValidation::analyze(edm::Event const& event, edm::EventSetup const& c) {
-  if (imc != 0) {
-    edm::Handle<edm::HepMCProduct> evtMC;
-    event.getByToken(tok_evt_, evtMC);  // generator in late 310_preX
-    if (!evtMC.isValid()) {
-      edm::LogWarning("Module named pfclusterAnalyzer") << "no HepMCProduct found";
-    }
+  double eta_MC = 0.;
+  double phi_MC = 0.;
+  double energy_MC = 0.;
+
+  edm::Handle<edm::HepMCProduct> hepMC;
+  event.getByToken(hepMCTok_, hepMC);
+  if (not hepMC.isValid()) {
+    edm::LogWarning("PFClusterValidation") << "HepMCProduct not found";
+    return;
+  }
 
-    // MC particle with highest pt is taken as a direction reference
-    double maxPt = -99999.;
-    int npart = 0;
-    const HepMC::GenEvent* myGenEvent = evtMC->GetEvent();
-    for (HepMC::GenEvent::particle_const_iterator p = myGenEvent->particles_begin(); p != myGenEvent->particles_end();
-         ++p) {
-      double phip = (*p)->momentum().phi();
-      double etap = (*p)->momentum().eta();
-      double pt = (*p)->momentum().perp();
-      double energy = (*p)->momentum().e();
-      if (pt > maxPt) {
-        npart++;
-        maxPt = pt;
-        energy_MC = energy;
-        phi_MC = phip;
-        eta_MC = etap;
-      }
+  // MC particle with highest pt is taken as a direction reference
+  double maxPt = -99999.;
+  const HepMC::GenEvent* myGenEvent = hepMC->GetEvent();
+  for (auto p = myGenEvent->particles_begin(); p != myGenEvent->particles_end(); ++p) {
+    double phip = (*p)->momentum().phi();
+    double etap = (*p)->momentum().eta();
+    double pt = (*p)->momentum().perp();
+    double energy = (*p)->momentum().e();
+    if (pt > maxPt) {
+      maxPt = pt;
+      energy_MC = energy;
+      phi_MC = phip;
+      eta_MC = etap;
     }
   }
-  Egen_MC->Fill(energy_MC);
+
+  egen_MC_->Fill(energy_MC);
 
   edm::Handle<reco::PFClusterCollection> pfClusterECAL;
-  event.getByToken(PFClusterECALTok_, pfClusterECAL);
-  //  reco::PFClusterCollection::const_iterator pf;
+  event.getByToken(pfClusterECALTok_, pfClusterECAL);
 
   edm::Handle<reco::PFClusterCollection> pfClusterHCAL;
-  event.getByToken(PFClusterHCALTok_, pfClusterHCAL);
+  event.getByToken(pfClusterHCALTok_, pfClusterHCAL);
 
   edm::Handle<reco::PFClusterCollection> pfClusterHO;
-  event.getByToken(PFClusterHOTok_, pfClusterHO);
+  event.getByToken(pfClusterHOTok_, pfClusterHO);
 
   edm::Handle<reco::PFClusterCollection> pfClusterHF;
-  event.getByToken(PFClusterHFTok_, pfClusterHF);
+  event.getByToken(pfClusterHFTok_, pfClusterHF);
 
-  double Econe = 0.;
-  double Hcone = 0.;
-  double HFcone = 0.;
-  double HOcone = 0.;
-
-  // sum the energy in a dr cone for each subsystem
-  Econe = sumEnergy(pfClusterECAL);
-  Hcone = sumEnergy(pfClusterHCAL);
-  HOcone = sumEnergy(pfClusterHO);
-  HFcone = sumEnergy(pfClusterHF);
+  // sum the energy in a dR cone for each subsystem
+  double Econe = sumEnergy(pfClusterECAL, eta_MC, phi_MC);
```

Comment:
```suggestion
  const double Econe = sumEnergy(pfClusterECAL, eta_MC, phi_MC);
```

Reply:
do these make any difference? the compiler can see that they are not being changed.

Reply:
It makes it clear to future developers that the values are not supposed to be changed